protobuf/shared.proto

```diff
@@ -64,6 +64,11 @@ message QueryReference {
   bytes fingerprint = 3;
 }
 
+message QueryPlanReference {
```

Comment:
Technically this could be in the `full_snapshot.proto` file, since we don't use these in compact snapshots (yet), but I think its also fine to have here.

Reply:
Yeah good point, I did debate about it. I think I added here because most of the references are in `shared.proto` even when it's only used with full snapshot (like FunctionReference, IndexReference). I'll keep this here for that reason.